### DISCUSSION TITLE
Android build/Linking issue: PackageList.java: error: package undefined does not exist import undefined.MyPackage;

### DISCUSSION THREAD

**Discussion by Tom-Bury** [callstack/react-native-builder-bob#428] [Q&A] [answered]
Hi!

I'm facing an issue with the plain starter template for a Native UI component on Android. When including the library (generated with the latest version) in my project (running RN 68.6) I'm getting the following error:

```
.../android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:55: error: package undefined does not exist
import undefined.MyPackage;
                ^
.../android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:175: error: cannot find symbol
      new MyPackage(),
          ^
  symbol:   class MyPackage
  location: class PackageList
```

I did not do any special steps on linking the library, so it should be fully using autolinking. When looking in Android Studio, the native library code seems to be included fine. It's only at build time that it seems to link it incorrectly.

I tested a plain starter library in fresh new RN app as well and that has no issues. So it might just be related to the older RN version used in my app, which I can't yet upgrade.

Any ideas on what could cause this issue & how I could resolve it?

Thanks!

**Comment by atlj** [comment]
I don't have any particular ideas but can you create a new empty library and try adding it to your app? It would be useful to know the root cause of the issue. It's should be either the template or the library.

**Reply by Tom-Bury** [reply]
Thanks for the quick response 😄 

I've spun up a quick test repo which consists of a plain library setup & two plain RN apps. One on the latest RN version, the other on 0.68.6. In there I'm facing the same issue: the 68.6 RN app fails to build on Android:

```
.../test-create-react-native-library/AwesomeOldProject/android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:15: error: package undefined does not exist
import undefined.TestCreateReactNativeLibraryPackage;
                ^
.../test-create-react-native-library/AwesomeOldProject/android/app/build/generated/rncli/src/main/java/com/facebook/react/PackageList.java:61: error: cannot find symbol
      new TestCreateReactNativeLibraryPackage()
```

You can find the repo here: <https://github.com/Tom-Bury/test-create-react-native-library>
The base steps in generating it were:
1. Generate the plain Native UI library
2. Pack that
3. Generate 2 RN testapps using `npx react-native@X.XX.X init AwesomeProject --version X.XX.X`
4. Include the tarball in both & add it as a dependency
5. Use the library view in both `App.tsx`
6. In each, try running `yarn run android` -> the new RN version runs fine, the old one fails

**Reply by atlj** [reply]
No problem 😅. Seems like it's an issue in the template itself then. I cannot turn this discussion into an issue. Could you create an issue with the same content?

**Reply by Tom-Bury** [reply]
Done! <https://github.com/callstack/react-native-builder-bob/issues/429>

I'll close this discussion then. Thanks for the help!

**Comment by atlj** [comment, marked as answer]
Please see https://github.com/callstack/react-native-builder-bob/issues/429#issuecomment-1655693378 for a temporary answer.
